Document SSO integration with Headscale and authentik (#994)
mother-of-all-self-hosting/mash-playbook: docs/services/headscale.md

########################################################################
```

### Single-Sign-On (SSO) integration

> Nextcloud supports Single-Sign-On (SSO) via OIDC. To make use of it, an Identity Provider (IdP) like [authentik](authentik.md), [authelia](authelia.md) or [tinyauth](tinyauth.md) needs to be set up.
> As Headscale's built in authentication is somewhat manual, setting up ODIC can provide a smoother user experience.

Reviewer: ```suggestion
Nextcloud supports Single-Sign-On (SSO) via OIDC. To make use of it, an Identity Provider (IdP) like [authentik](authentik.md), [Authelia](authelia.md) or [Tinyauth](tinyauth.md) needs to be set up.

As Headscale's built in authentication is somewhat manual, setting up ODIC can provide a smoother user experience.
```

For example, you can enable SSO with authentik via OIDC by following the steps below:

> Here, we are using ansible vault to supply both our `domain` as well as `client_id` and `client_secret`, you would add the following configuration to your vars.yml. 
> This assume that you pick the slug `headscale` in Authentik when adding headscale as an application in Authentik. If not, replace `headscale` in `issuer: "https://authentik.{{ domain }}/application/o/headscale/"`

Reviewer: ```suggestion
Here, we are using Ansible vault to supply both our `domain` as well as `client_id` and `client_secret`, you would add the following configuration to your `vars.yml` file. This assume that you have picked the slug `headscale` in authentik when adding Headscale as an application in authentik. If not, replace `headscale` in `issuer: "https://authentik.{{ domain }}/application/o/headscale/"`
```

```yaml
headscale_configuration_extension:
  oidc:
    only_start_if_oidc_is_available: true
    issuer: "https://authentik.{{ domain }}/application/o/headscale/"
    client_id: "{{ vault_headscale_client_id }}"
    client_secret: "{{ vault_headscale_client_secret }}"
    scope: ["openid", "profile", "email"]
    pkce:
      enabled: true
      method: S256
```

> You can find more details reading about configuring OIDC by referring to the documentation at both [Headscale](https://headscale.net/stable/ref/oidc/?h=oidc) and [Authentik](https://integrations.goauthentik.io/networking/headscale/ ).
> Note that Headscale's documentation doesn't explicitly cover Authentik. 

Reviewer: ```suggestion

You can find more details about configuring OIDC by referring to the documentation at both [Headscale](https://headscale.net/stable/ref/oidc/?h=oidc) and [authentik](https://integrations.goauthentik.io/networking/headscale/ ). Note that Headscale's documentation doesn't explicitly cover authentik. 
```
